### PR TITLE
gPay Logging Expectation Changes

### PR DESCRIPTION
**Pull request by mborof**
This repairs the expectations when returning from gPay.  In order to determine whether a redirect occurred, we will now check the analytics service to see if the user was sent to gPay (if they weren't sent to gPay it means the user was redirected there because we lost JavaScript memory).  

This replaces a "had ever logged" flag which will not work anymore due to the introduction of a "page load" log event.

### CODE REVIEW

File: src/runtime/analytics-service.js [hunk 1]
```diff
@@ -322,22 +327,31 @@ export class AnalyticsService {
...
     if (
       ClientEventManager.isPublisherEvent(event) &&
       !this.shouldLogPublisherEvents_() &&
       !this.shouldAlwaysLogEvent_(event)
     ) {
       return;
     }
+
+    switch (event.eventType) {
+      //this event is just used to communicate information internally.  It should
```
Comment (henel677) [inline src/runtime/analytics-service.js]: // This...

File: src/runtime/analytics-service.js [hunk 2]
```diff
@@ -322,22 +327,31 @@ export class AnalyticsService {
...
+      //not be reported to the SwG analytics service.
+      case AnalyticsEvent.EVENT_SUBSCRIPTION_STATE:
+        return;
+      case AnalyticsEvent.ACTION_PAYMENT_FLOW_STARTED:
+        if (this.waitingOnGpay_) {
+          //TODO: Consider a permanent log event for this
```
Comment (henel677) [inline src/runtime/analytics-service.js]: // TODO

File: src/runtime/analytics-service.js [hunk 3]
```diff
@@ -267,11 +267,16 @@ export class AnalyticsService {
...
+   * Returns true if the user was sent to gPay (and not redirected).
    * @return {boolean}
    */
-  getHasLogged() {
-    return this.everStartedLog_;
+  getWaitingOnGpay() {
+    return this.waitingOnGpay_;
+  }
+
+  /** */
```
Comment (henel677) [inline src/runtime/analytics-service.js]: Do you need this line? If you do It is nicer if you add some description even if it is mostly restating what can be deduced from the name of the function

File: src/runtime/pay-flow.js [hunk 4]
```diff
@@ -389,6 +392,7 @@ function validatePayResponse(deps, payPromise, completeHandler) {
       }
     }
     deps.eventManager().logSwgEvent(eventType, true, eventParams);
+    deps.analytics().setDoneWaitingOnGpay();
```
[the remark this answers is not in the document]
Comment (mborof) [reply src/runtime/pay-flow.js]: I am moving this dev to a new branch where I am making the pay logic self reliant (and removing this reliance on the logging system).  I did account for this comment in that new branch though.  Good idea!

**Pull request closed without merging**